### ISSUE TITLE
box.delete(key), data really delete?

### ISSUE THREAD

**Issue by weishirongzhen** [hivedb/hive#793]
**Question**
When call box.delete(key), data isn't really delete from disk?

**Code sample**
```dart
@HiveType(typeId: 0)
class TextModel extends HiveObject {
TextModel(this.address);
    @HiveField(0)
    late String address;
}
 
final keyForTest = await box.add(TextModel("abc123"));
box.delete(keyForTest);


```

restart app, then print log in .g.dart file, TextModelAdapter in read function 
```dart

 TextModel read(BinaryReader reader) {
    final numOfFields = reader.readByte();
    final fields = <int, dynamic>{
      for (int i = 0; i < numOfFields; i++) reader.readByte(): reader.read(),
    };
    print(fields[0]);   //even when box is empty,    print(fields[0])  still print old data  "abc123" twice, why?
    return TextModel()
      
      ..address = fields[0] as String
    
  }

```

**Version**
 - Platform: iOS, Android
 - Flutter version: 2.5.0
 - Hive version: 2.0.4

**Comment by mbaumgartenbr** [comment]
Hey @weishirongzhen!

TL;DR Boxes need to be compacted, see:

https://docs.hivedb.dev/#/advanced/compaction

**Comment by weishirongzhen** [comment]
> Hey @weishirongzhen!
> 
> TL;DR Boxes need to be compacted, see:
> 
> https://docs.hivedb.dev/#/advanced/compaction

Thanks, I'll look into it
